pkg/ctl/install/fluxinstaller.go

```diff
@@ -0,0 +1,558 @@
+	if opts.gitURL == "" {
+		return nil, fmt.Errorf("please supply a valid --git-url argument")
+	}
+	if opts.gitEmail == "" {
+		return nil, fmt.Errorf("please supply a valid --git-email argument")
+	}
+	helmOpTLS := true
+	if !opts.noHelmOp && opts.noTiller {
+		// check that either all or none of the TLS files are provided
+		tlsPaths := []string{opts.helmOpTLSKeyFile, opts.helmOpTLSCertFile, opts.helmOpTLSKeyFile}
+		numTLSPaths := len(tlsPaths)
+		numNonEmptyTLSFiles := 0
+		for _, path := range tlsPaths {
+			if len(path) > 0 {
+				numNonEmptyTLSFiles++
+			}
+		}
+		if numNonEmptyTLSFiles == 0 {
+			helmOpTLS = false
+		} else if numNonEmptyTLSFiles != numTLSPaths {
+			return nil, fmt.Errorf("either none or all --helmop-tls-*-path flags should be provided")
+		}
+	}
+
+	if err := cmdutils.NewMetadataLoader(cmd).Load(); err != nil {
+		return nil, err
+	}
+	cfg := cmd.ClusterConfig
+	ctl, err := cmd.NewCtl()
+	if err != nil {
+		return nil, err
+	}
+
+	if err := ctl.CheckAuth(); err != nil {
+		return nil, err
+	}
+	if err := ctl.RefreshClusterConfig(cfg); err != nil {
+		return nil, err
+	}
+	kubernetesClientConfigs, err := ctl.NewClient(cfg)
+	if err != nil {
+		return nil, err
+	}
+	k8sConfig := kubernetesClientConfigs.Config
+
+	k8sRestConfig, err := clientcmd.NewDefaultClientConfig(*k8sConfig, &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return nil, errors.Errorf("cannot create Kubernetes client configuration: %s", err)
+	}
+	k8sClientSet, err := kubeclient.NewForConfig(k8sRestConfig)
+	if err != nil {
+		return nil, errors.Errorf("cannot create Kubernetes client set: %s", err)
+	}
+
+	gitClient := git.NewGitClient(ctx, opts.gitUser, opts.gitEmail, opts.timeout)
+	fi := &fluxInstaller{
+		opts:          opts,
+		helmOpTLS:     helmOpTLS,
+		cmd:           cmd,
+		k8sConfig:     k8sConfig,
+		k8sRestConfig: k8sRestConfig,
+		k8sClientSet:  k8sClientSet,
+		gitClient:     gitClient,
+	}
+	return fi, nil
+}
+
+func (fi *fluxInstaller) run(ctx context.Context) error {
+	pki, pkiPaths, err := fi.setupPKI()
+	if err != nil {
+		return err
+	}
+
+	logger.Info("Generating manifests")
+	manifests, secrets, err := fi.getManifestsAndSecrets(pki, pkiPaths)
+	if err != nil {
+		return err
+	}
+
+	logger.Info("Cloning %s", fi.opts.gitURL)
+	cloneDir, err := fi.gitClient.CloneRepo("eksctl-install-flux-clone-", fi.opts.gitBranch, fi.opts.gitURL)
+	if err != nil {
+		return fmt.Errorf("cannot clone repository %s: %s", fi.opts.gitURL, err)
+	}
+	cleanCloneDir := false
+	defer func() {
+		if cleanCloneDir {
+			_ = fi.gitClient.DeleteLocalRepo()
+		} else {
+			logger.Critical("You may find the local clone of %s used by eksctl at %s",
+				fi.opts.gitURL,
+				cloneDir)
+		}
+	}()
+
+	logger.Info("Writing Flux manifests")
+	fluxManifestDir := filepath.Join(cloneDir, fi.opts.gitFluxPath)
+	if err := writeFluxManifests(fluxManifestDir, manifests); err != nil {
+		return err
+	}
+
+	if fi.opts.amend {
```

Comment:
What is this for?

Reply:
To stop for amending. Please read the parameter description.

Reply:
That is understood, I am questioning the purpose. Is it being assumed that user may wish to have that options?

Reply:
unrelated to this PR, this was introduced in #1064